### DISCUSSION TITLE
Picamera2 MJPEG Streaming Demo vs direct stream exposure differences

### DISCUSSION THREAD

**Discussion by vulterey** [raspberrypi/picamera2#957] [General]
Hi All,

I am using mjpeg_server_2.py to get a live stream from the Odseven Raspberry Pi Camera Board - "Spy Camera" (5MP). 
I have another flask app which produces an HTML page.
To get the live stream on this webpage I am using the <iframe> tag to display the direct stream: http://192.168.1.10:8080/stream.mjpg.
The problem is that there is a significant difference in exposure (I guess) between the direct stream (2.jpg - darker picture) and HTML streaming webpage http://192.168.1.10:8080/index.html (1.jpg - lighter picture)
I tried using picam2.set_controls({"ExposureTime":10000, "AnalogueGain": 1.0}) but this only caused totally black view across both ways of streaming. I tried different ExposureTime values without any luck.
![2](https://github.com/raspberrypi/picamera2/assets/44617758/5597db1e-2666-4157-9e6b-5470513f8f11)
![1](https://github.com/raspberrypi/picamera2/assets/44617758/737653b5-0840-4102-ae1a-d61ff8cd8b4c)

Question:
Is there any way to increase the exposure of the direct mjpg stream?

Many thanks for help in advance
Adam

**Comment by sandyol55** [comment]
As the example uses a 'create_video_configuration', the default framerate will be 30 fps, so forcing a wider range of allowable frame durations might allow for a longer exposure. Also forcing a mild 'overexposure' might help .
Both of these can be applied with a controls setting of 

```
picam2.set_controls({"FrameDurationLimits":(33333,100000),"ExposureValue":3.0})

```

Don't think this will influence only one stream but might drive the video levels up so the difference is less noticeable???

Edit.
And if you really want to push the video levels up even more,  you can add some 'Brightness' too.
```
picam2.set_controls({"FrameDurationLimits":(33333,100000),"ExposureValue":5.0, "Brightness":0.2}

```

**Reply by vulterey** [reply]
Hi sandyol55,

Thank you very much for the reply and the code.
It helps a lot and indeed makes things brighter. 

The funniest thing is that the lack of common sense skewed my observation. 

There is no difference between direct stream and stream displayed in HTML.
I thought the stream in HTML was brighter because when I was switching between the windows and had the camera pointed to my face the white background in the HTML tab acted as an additional source of light making my face brighter.
When I was looking at the direct stream tab with a black background the additional light from HTML was gone making my face look darker. I still cannot believe I didn't notice it 🤦 

I am living this here for future generations: 
When you are stuck with something, stop trying to blame everything else except yourself. 
Step back and come back the next day otherwise, you will make a mistake and look stupid 😁 

Please close this embarrassing case 😁 😁 😁

**Reply by sandyol55** [reply]
We've all done something similar - or at least I know I have!
Good 'advice to self' in there, must try and remember it.

**Reply by vulterey** [reply]
Thank you again for your help.
